sklearn/feature_extraction/tests/test_text.py

```diff
@@ -512,12 +512,16 @@ def test_tfidf_vectorizer_setters():
     tv = TfidfVectorizer(norm='l2', use_idf=False, smooth_idf=False,
                          sublinear_tf=False)
     tv.norm = 'l1'
+    tv.fit(ALL_FOOD_DOCS)
     assert tv._tfidf.norm == 'l1'
     tv.use_idf = True
+    tv.fit(ALL_FOOD_DOCS)
     assert tv._tfidf.use_idf
     tv.smooth_idf = True
+    tv.fit(ALL_FOOD_DOCS)
```

Comment:
should we invalidate/remove `_tfidf` whenever a parameter is changed here?

Reply:
I would say no. In our estimators, if a user changes any of the parameters, the underlying private attributes do not update or get removed (unless they call fit again).

xref: https://github.com/scikit-learn/scikit-learn/issues/17158